userservice/src/main/java/ro/devdepot/services/implementation/UserServiceImpl.java

```diff
@@ -27,4 +33,42 @@ public LoginResponse login(LoginRequest loginRequest) {
 
         return null;
     }
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User updateUser(User user, Long id) {
+        User existingUser = userRepository.findById(id).orElse(null);
+
+        existingUser.setUsername(user.getUsername());
+
+        return userRepository.save(existingUser);
+    }
+
+    @Override
+    public User updateUserRole(UserRole newuserRole, Long id) {
```

Comment:
this method should not be used by the user so you should add another parameter called loggedInUser to validate if the user that is using it is an admin, but that comes later, for now delete the entire method